lib/vector-common/src/finalizer.rs

```diff
@@ -115,7 +115,7 @@ where
             tokio::select! {
                 biased;
                 _ = &mut shutdown, if handle_shutdown => break,
-                _ = flush.notified() => {
+                () = flush.notified() => {
```

Comment:
This seems like an odd thing to have to change for the AWS updates 🤔 

aside from questioning it in general I'm also curious what the difference manifests as?

Reply:
I'll revert. It was likely unintentional.